plugins/languages/treesitter/treesitter.nix

```diff
@@ -12,6 +12,101 @@ helpers.neovim-plugin.mkNeovimPlugin config {
   luaName = "nvim-treesitter.configs";
   defaultPackage = pkgs.vimPlugins.nvim-treesitter;
 
+  description = ''
+    Provides an interface to [tree-sitter]
+
+    ## Installing Your Own Grammars with Nixvim
```

Comment:
Will need to check using our `serve-docs` devshell command, but I have a feeling mdbook doesn't support markdown headers within the option descriptions. The plugin description may work differently though, not sure.

Reply:
![image](https://github.com/user-attachments/assets/b217830f-41ce-46b7-9b5b-1c11dd5b78bd)

Reply:
Awesome, thanks for checking! We could debate which heading level to use, but I don't want to get too nit-picky! (I'd probably prefer `###` or `####`, but wouldn't like to say without checking how it renders).

Reply:
Looks better with `###`. One thing I'm noticing is that it gets rid of the plugin doc link and maintainer link with a description. We might want to fix that.
![image](https://github.com/user-attachments/assets/af630b0e-1532-4d8e-a9e8-b9cf0903cffd)

Reply:
>  One thing I'm noticing is that it gets rid of the plugin doc link and maintainer link with a description. We might want to fix that.

Nvm, I'll create a PR. I see what it's doing.